exporters/dsexporter/dsexporter.go

```diff
@@ -1,39 +1,115 @@
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxProbeSuccess: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_probe_success",
+			Help: "Availability of konflux service/component",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.konfluxProbeSuccess.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	var availability float64
+
+	availability = IsDataSourceExist(GetDataSources(GetGrafanaResource()), check)
+	e.konfluxProbeSuccess.WithLabelValues(service, check).Set(availability)
+	e.konfluxProbeSuccess.Collect(ch)
+}
+
+// get the grafna resource as a map
+func GetGrafanaResource() map[string]interface{} {
+	clientset := NewKubeClient()
+
+	data, err := clientset.RESTClient().
+		Get().
+		AbsPath("/apis/grafana.integreatly.org/v1beta1").
+		Namespace("appstudio-grafana").
+		Resource("grafanas").
+		Name("grafana-oauth").
+		DoRaw(context.TODO())
+	var grafanaResoruce map[string]interface{}
+	err = json.Unmarshal(data, &grafanaResoruce)
+	if err != nil {
+		fmt.Printf("Error getting resource: %v\n", err)
+		os.Exit(1)
+	}
+
+	return grafanaResoruce
+}
+
+// get datasources from grafana resource
+func GetDataSources(grafanaResource map[string]interface{}) []string {
+	// return empty string slice if datasources are not defined
+	if grafanaResource["status"].(map[string]any)["datasources"] == nil {
+		return make([]string, 0)
+	}
+	datasourcesIfc := grafanaResource["status"].(map[string]any)["datasources"].([]interface{})
+	datasources := make([]string, len(datasourcesIfc))
+	for i, v := range datasourcesIfc {
+		datasources[i] = v.(string)
+	}
+	return datasources
+}
+
+// check if datasource exists, return 1 if yes, 0 if not
+func IsDataSourceExist(datasources []string, dsToCheck string) float64 {
+	for _, datasource := range datasources {
+		if strings.Contains(datasource, dsToCheck) {
+			fmt.Println("Datasource", datasource, "exists")
+			return 1
+		}
+	}
+	fmt.Println("Datasource", dsToCheck, "does not exist")
+	return 0
+}
+
+func NewKubeClient() *kubernetes.Clientset {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
```

Comment:
Have a look at [GetConfigOrDie](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/config#GetConfigOrDie). I think it does what you're doing here.